BookReader/plugins/plugin.tts.js

```diff
@@ -1,527 +1,2 @@
(BookReader.prototype.buildMobileDrawerElement),BookReader.prototype.initNavbar=function(t){return function(){var e=t.call(this);return this.options.enableTtsPlugin&&this.ttsEngine.isSupported&&$("<button class='BRicon read js-tooltip'></button>").insertAfter(e.find(".BRpage .BRicon.thumb")),e}}(BookReader.prototype.initNavbar),BookReader.prototype.ttsToggle=function(){this.autoStop&&this.autoStop(),this.ttsEngine.playing?this.ttsStop():this.ttsStart()},BookReader.prototype.ttsStart=function(){this.constModeThumb==this.mode&&this.switchMode(this.constMode1up),this.$(".BRicon.read").addClass("unread"),this.ttsEngine.start(this.currentIndex(),this.getNumLeafs())},BookReader.prototype.ttsStop=function(){this.$(".BRicon.read").removeClass("unread"),this.ttsEngine.stop(),this.ttsRemoveHilites(),this.removeProgressPopup()},BookReader.prototype.ttsBeforeChunkPlay=function(t){var e=this;return this.ttsMaybeFlipToIndex(t.leafIndex).then((function(){e.ttsHighlightChunk(t),e.ttsScrollToChunk(t)}))},BookReader.prototype.ttsMaybeFlipToIndex=function(t){var e=this.constMode2up==this.mode,n=null,o=new Promise((function(t){return n=t}));e?t==this.twoPage.currentIndexR||t==this.twoPage.currentIndexL?n():(this.animationFinishedCallback=n,this.next(),o.then(this.ttsMaybeFlipToIndex.bind(this,t))):(this.jumpToIndex(t),n());return o},BookReader.prototype.ttsHighlightChunk=function(t){this.ttsRemoveHilites(),this.constMode2up==this.mode?this.ttsHilite2UP(t):this.ttsHilite1UP(t)},BookReader.prototype.ttsScrollToChunk=function(t){if(this.constMode1up==this.mode){var e,n=0;for(e=0;e<t.leafIndex;e++)n+=parseInt(this._getPageHeight(e)/this.reduce)+this.padding;var o=t.lineRects[0][3],r=t.lineRects[t.lineRects.length-1][1],i=n+o/this.reduce,s=n+r/this.reduce;soundManager.debugMode&&console.log("leafTop = "+n+" topOfFirstChunk = "+i+" botOfLastChunk = "+s);var u=this.refs.$brContainer.prop("scrollTop"),a=u+this.refs.$brContainer.height();soundManager.debugMode&&console.log("containerTop = "+u+" containerBot = "+a),(i<u||s>a)&&this.refs.$brContainer.stop(!0).animate({scrollTop:i},"fast")}},BookReader.prototype.ttsHilite1UP=function(t){for(var e=0;e<t.lineRects.length;e++){var n=t.lineRects[e][0],o=t.lineRects[e][1],r=t.lineRects[e][2],i=t.lineRects[e][3],s=document.createElement("div");this.ttsHilites.push(s),$(s).prop("className","BookReaderSearchHilite").appendTo(this.$(".pagediv"+t.leafIndex)),$(s).css({width:(r-n)/this.reduce+"px",height:(o-i)/this.reduce+"px",left:n/this.reduce+"px",top:i/this.reduce+"px"})}},BookReader.prototype.ttsHilite2UP=function(t){for(var e=0;e<t.lineRects.length;e++){var n=t.lineRects[e][0],o=t.lineRects[e][1],r=t.lineRects[e][2],i=t.lineRects[e][3],s=document.createElement("div");this.ttsHilites.push(s),$(s).prop("className","BookReaderSearchHilite").css("zIndex",3).appendTo(this.refs.$brTwoPageView),this.setHilightCss2UP(s,t.leafIndex,n,r,i,o)}},BookReader.prototype.ttsRemoveHilites=function(){$(this.ttsHilites).remove(),this.ttsHilites=[]}}]);
```

Comment:
How come this file is minified?

Reply:
Mostly because that's the webpack default. I kind of want it to be minified, because then it's less likely someone will edit it by accident and not edit the original. Thoughts?

Reply:
These files are now no longer in the repo, so should be good.